File: 111. Minimum Depth of Binary Tree/111. Minimum Depth of Binary Tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
#define MIN(a,b) ((a) < (b) ? (a) : (b))

int minDepth(struct TreeNode* root) {
    if (!root) return 0;
    if (!root->left && !root->right) return 1;
    
    int leftDepth = root->left ? minDepth(root->left) : INT_MAX;
    int rightDepth = root->right ? minDepth(root->right) : INT_MAX;
    
    return 1 + MIN(leftDepth, rightDepth);
}

File: 111. Minimum Depth of Binary Tree/111. Minimum Depth of Binary Tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func minDepth(root *TreeNode) int {
    if root == nil {
        return 0
    }
    if root.Left == nil && root.Right == nil {
        return 1
    }
    
    leftDepth := math.MaxInt32
    rightDepth := math.MaxInt32
    
    if root.Left != nil {
        leftDepth = minDepth(root.Left)
    }
    if root.Right != nil {
        rightDepth = minDepth(root.Right)
    }
    
    return 1 + min(leftDepth, rightDepth)
}